pkg/executor/poolmgr/gpm.go

```diff
@@ -130,6 +132,41 @@ func (gpm *GenericPoolManager) Run(ctx context.Context) {
 	go gpm.idleObjectReaper()
 }
 
+func (gpm *GenericPoolManager) RecycleFuncPods(logger *zap.Logger, f fv1.Function) error {
+
+	env, err := gpm.fissionClient.Environments(f.Spec.Environment.Namespace).Get(f.Spec.Environment.Name)
+	if err != nil {
+		return err
+	}
+
+	gp, err := gpm.GetPool(env)
+	if err != nil {
+		return err
+	}
+
+	funcLabels := gp.labelsForFunction(&f.Metadata)
+
+	podList, err := gpm.kubernetesClient.CoreV1().Pods(metav1.NamespaceAll).List(metav1.ListOptions{
+		LabelSelector: labels.Set(funcLabels).AsSelector().String(),
+	})
+
+	if err != nil {
+		return err
+	}
+
+	for _, po := range podList.Items {
+		err := gpm.kubernetesClient.CoreV1().Pods(po.ObjectMeta.Namespace).Delete(po.ObjectMeta.Name, &metav1.DeleteOptions{})
```

Comment:
We also need to invalidate the cache so that executor will specialize a new pod for function.
One thing I worry about `deleting pod` here is when a configmap/secret is comsumed by many functions, once it get updated poolmanager will specialize tons of new pods, thus cluster runs up all available resource in the end if `old` pods are still in `Terminating` state.